docs/product-editor-development/README.md

```diff
@@ -0,0 +1,22 @@
+# Developing extensions for the product editor
+
+> ⚠️ **Notice:** This documentation is currently a **work in progress**. While it's open to the public for transparency and collaboration, please be aware that some sections might be incomplete or subject to change. We appreciate your patience and welcome any contributions!
+
+This handbook is a guide for extension developers looking to add support for the new product editor in their extensions. The product editor uses [Gutenberg's Block Editor](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-editor), which is going to help WooCommerce evolve alongside the WordPress ecosystem.
+
+The product editor's UI consists of Groups (currently rendered as tabs), Sections, and Fields, which are all blocks.
+
+![Product editor structure](https://woocommerce.files.wordpress.com/2023/09/block-product-editor-structure.png)
```

Comment:
It would be ideal if color wasn't the only differentiator used in this image, as it can make it difficult for some people.

Minor nit-pick... both List Price and Sale Price are their own fields.